### PR TITLE
fix(recordingoptions): client may explicitly unset options

### LINKED ISSUE #634: itest failures due to ordering, failed/lack of cleanup
We're currently having some flaky itest results with sporadic failures that seem to change frequently, so I suspect instability due to (intentionally, for this kind of reason) randomized test ordering and probably some missing cleanup somewhere, particularly if test failures like #633 may cause the associated cleanup to be skipped or also fail, causing more cascading failures elsewhere.

`repeated-integration-tests.sh` will probably be helpful to track down these various failures.

- [ ] #632 
- [ ] #636

### PR DESCRIPTION
**Pull request by andrewazores**
Related to #632
Fixes #634

### CODE REVIEW

**Comment by hareetd** [comment]
Out of curiosity, how exactly was not being able to explicitly unset the recording options leading to the test failures from before?

**Comment by andrewazores** [comment]
> Out of curiosity, how exactly was not being able to explicitly unset the recording options leading to the test failures from before?

This seems to have been the key: https://github.com/cryostatio/cryostat/pull/640/files#diff-5191e54db18231ddfc4a99eb2dd6289336bc68561507563c9c471d4f8986b0d7L74

When retrieving the options from the JVM, the default value being unset is reported as 0. But, explicitly setting a value and then "resetting" it to 0 seems to turn out to actually tell the target JVM to not record anything at all, as in the recording may have a maximum size of 0 bytes.

There's probably a better/correct way to get the `GET` recordingOptions data from the JMC API we're using to retrieve this so that we can return data to the client in the first place that distinguishes unset options from options set to 0/false, too.

**Review by hareetd — DISMISSED** [review]
Looks good.

File: src/test/java/io/cryostat/net/web/http/api/v1/TargetRecordingOptionsPatchHandlerTest.java [hunk 1]
```diff
@@ -159,6 +159,52 @@ public Map answer(InvocationOnMock args) throws Throwable {
         }
     }
 
+    @Test
+    void shouldUnsetRecordingOptions() throws Exception {
+        Map<String, String> defaultValues =
+                Map.of("toDisk", "true", "maxAge", "50", "maxSize", "32");
```
Comment (jan-law) [inline src/test/java/io/cryostat/net/web/http/api/v1/TargetRecordingOptionsPatchHandlerTest.java:165]: Where do these default values come from (unless the values are arbitrary)?

**Review by jan-law — COMMENTED** [review]
Besides this last question, looks good

File: src/test/java/io/cryostat/net/web/http/api/v1/TargetRecordingOptionsPatchHandlerTest.java (hunk 1 above)
Comment (andrewazores) [reply src/test/java/io/cryostat/net/web/http/api/v1/TargetRecordingOptionsPatchHandlerTest.java:165]: These are actually just arbitrary for this test. I should rename that variable.

**Review by hareetd — APPROVED** [review]

**Review by jan-law — APPROVED** [review]

**Pull request merged**